aten/src/ATen/SparseTensorImpl.h

```diff
@@ -182,6 +189,17 @@ struct CAFFE2_API SparseTensorImpl : public TensorImpl {
   // make it happen
   void set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values);
 
+  c10::intrusive_ptr<TensorImpl> shallow_copy_and_detach() const override {
+    auto impl = c10::make_intrusive<SparseTensorImpl>(type_id(), dtype());
+    impl->size_ = sizes().vec();
```

Comment:
is there a reason we haven't moved this to just be sizes_ yet?

Reply:
IMO you should also copy literally everything, even strides_, is_contiguous_, is_wrapped_number, etc.

Reply:
For removing `size_` and just use `sizes_`, when working on https://github.com/pytorch/pytorch/pull/11259 I ran into some test failures that don't seem to be related to that PR, and haven't fully investigated why it happens. The current plan is to do it after the `data_` field is removed from Variable::Impl (the 6th PR mentioned in https://github.com/pytorch/pytorch/issues/13638).